Support httpclient5.x for tag transmission

SunJiFengPlus opened the pull request:
What type of PR is this?

Feature

What this PR does / why we need it?

Describe it.
Support httpclient5.x for tag transmission

Which issue(s) this PR fixes？

Fixes #1784

Does this PR introduce a user-facing change?

No

<!-- Follow this checklist to help us incorporate your contribution quickly and easily: -->

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [ ] GitHub Actions works fine in this PR.

SunJiFengPlus committed:
Support httpclient5.x for tag transmission

Signed-off-by: sunjifeng <jifeng.sun@outlook.com>

provenceee commented on sermant-plugins/sermant-tag-transmission/tag-transmission-httpclient5.x-plugin/src/main/java/io/sermant/tag/transmission/httpclientv5/interceptors/HttpClient5Interceptor.java:
```diff
@@ -0,0 +1,92 @@
+                continue;
+            }
+            for (String value : values) {
+                httpRequest.addHeader(key, value);
```
If the request header exists, do not add it to prevent duplicate addition.

hanbingleixue commented on .github/actions/common/plugin-change-check/action.yml:
```diff
@@ -311,6 +311,25 @@ runs:
+    - name: env tag-transmission-httpclientV5
+      shell: bash
+      run: |
+        echo "httpclientV4TagTransmissionChanged=${{ steps.changed-tag-transmission-httpclientV5.outputs.changed }}" >> $GITHUB_ENV
```
It should be httpclientV5TagTransmissionChanged

hanbingleixue commented on .github/actions/scenarios/tag-transmission/httpclientv5/action.yml:
```diff
@@ -0,0 +1,46 @@
+    - name: package demos
+      shell: bash
+      run: |
+        mvn package -Dhttpclient4x.version=${{ matrix.httpclientVersion }} -DskipTests -Phttpclientv5-test --file \
```
It should be httpclient5x.version

hanbingleixue commented on .gitignore:
```diff
@@ -8,8 +8,9 @@
 
 # build artifacts
-sermant-agent-*
+sermant-agent*
```
It cannot be changed to sermant-agent*. If it is changed to sermant-agent*, the change of sermant-agentcore  will be ignored.

SunJiFengPlus committed:
Modified according to review suggestions
1. Fix the wrong version in GitHub action
2. Delete redundant files in .gitignore

Signed-off-by: sunjifeng <jifeng.sun@outlook.com>

SunJiFengPlus commented on sermant-integration-tests/tag-transmission-test/httpclientv5-demo/pom.xml:
```diff
@@ -0,0 +1,46 @@
+
+    <artifactId>httpclientv5-demo</artifactId>
+
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+        <httpcore.version>5.3.4</httpcore.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents.client5</groupId>
+            <artifactId>httpclient5</artifactId>
+            <version>${httpclient5x.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents.core5</groupId>
+            <artifactId>httpcore5</artifactId>
+            <version>${httpcore.version}</version>
+        </dependency>
+    </dependencies>
```
In httpclient repository 5.4.x depends on httpcore5-5.3.4, but in the maven repository depends on version 5.1.5, which will cause `NoSuchMethodError`  at runtime, so I manually adjusted the version of httpcore5, otherwise it will not run, I guess that other users of httpclient5.4,5.5 will have the same problem.

https://github.com/apache/httpcomponents-client/blob/rel/v5.4.4/pom.xml#L65

```bash
httpclientv5-demo % mvn dependency:tree | grep -A 4 'httpclient5'
[INFO] \- org.apache.httpcomponents.client5:httpclient5:jar:5.4.4:compile
[INFO]    +- org.apache.httpcomponents.core5:httpcore5:jar:5.1.5:compile
[INFO]    +- org.apache.httpcomponents.core5:httpcore5-h2:jar:5.1.5:compile
[INFO]    \- org.slf4j:slf4j-api:jar:1.7.36:compile
[INFO] ------------------------------------------------------------------------
```

lilai23 replied:
OK，it is necessasy to notice this
